File: 2015/01.cc
#include <iostream>
#include <fstream>
#include <string>

int main()
{
  std::string line;
  std::ifstream file("1_input.txt");
  file >> line;
  file.close();
  
  int floor = 0, line_cnt = 1;
  for (auto c : line) {
    if (c == '(') {
      floor++;
    } else {
      floor--;
    }

    if (floor == -1) {
      std::cout << line_cnt << "\n";
    }

    line_cnt++;
  }
  
  std::cout << floor << "\n";
  return 0;
}

File: 2018/01.pl
#!/usr/bin/env perl
use strict;
use warnings;

open my $input, "<", "01_input.txt" or die "Can't open input file\n";

my $frequency = 0;
while (my $line = <$input>) {
  chomp $line;  
  $frequency += $line;
}

print "Answer: $frequency\n";
close $input;
